client/src/app/controllers/search-project-controller/search-project.service.ts

```diff
@@ -0,0 +1,89 @@
+import { Injectable } from '@angular/core';
+import { HttpClient } from '@angular/common/http';
+import { Observable } from 'rxjs';
+import { AuthService } from '../auth-controller/auth.service';
+import { environment } from 'src/environments/environment';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class SearchProjectService {
+  apiUrl = environment.apiUrl;
+
+  constructor(private http: HttpClient, private authService: AuthService) { }
+  //This method is pretty simple, just fill in the fields that exist or will be used with the search or null if not. To the best of my knowledge, js/ts does 
+  //not support named parameters so if you wanted to only serach with deadline, the correct call to this method would be searchProjectsMultipleParams(null, null, null, null, null, null, DATE PARAMETER)
```

Comment:
This is nitpicky, but as a general rule, please don't make comments longer than the maximum of (80 characters, the length of the longest line of code). You don't have to change your existing comments, but please try to not make them excessively long in the future.